src/main/java/org/cdc/gov/sdp/aphl/AphlS3Producer.java

```diff
@@ -0,0 +1,334 @@
+import com.amazonaws.services.s3.model.PutObjectResult;
+import com.amazonaws.services.s3.model.StorageClass;
+import com.amazonaws.services.s3.model.UploadPartRequest;
+
+/**
+ * A Producer which sends messages to the Amazon Web Service Simple Storage
+ * Service <a href="http://aws.amazon.com/s3/">AWS S3</a>
+ */
+public class AphlS3Producer extends DefaultProducer {
+
+	private static final Logger LOG = LoggerFactory.getLogger(AphlS3Producer.class);
+
+	private transient String s3ProducerToString;
+
+	public AphlS3Producer(final Endpoint endpoint) {
+		super(endpoint);
+	}
+
+	@Override
+	public void process(final Exchange exchange) throws Exception {
+
+		if (getConfiguration().isMultiPartUpload()) {
+			processMultiPart(exchange);
+		} else {
+			processSingleOp(exchange);
+		}
+	}
+
+	public void processMultiPart(final Exchange exchange) throws Exception {
+		File filePayload = null;
+		Object obj = exchange.getIn().getMandatoryBody();
+		// Need to check if the message body is WrappedFile
+		if (obj instanceof WrappedFile) {
+			obj = ((WrappedFile<?>) obj).getFile();
+		}
+		if (obj instanceof File) {
+			filePayload = (File) obj;
+		} else {
+			throw new InvalidArgumentException("aphl-s3: MultiPart upload requires a File input.");
+		}
+
+		ObjectMetadata objectMetadata = determineMetadata(exchange);
+		if (objectMetadata.getContentLength() == 0) {
+			objectMetadata.setContentLength(filePayload.length());
+		}
+
+		final String keyName = determineKey(exchange);
+		final InitiateMultipartUploadRequest initRequest = new InitiateMultipartUploadRequest(
+				getConfiguration().getBucketName(), keyName, objectMetadata);
+
+		String storageClass = determineStorageClass(exchange);
+		if (storageClass != null) {
+			initRequest.setStorageClass(StorageClass.fromValue(storageClass));
+		}
+
+		String cannedAcl = exchange.getIn().getHeader(S3Constants.CANNED_ACL, String.class);
+		if (cannedAcl != null) {
+			CannedAccessControlList objectAcl = CannedAccessControlList.valueOf(cannedAcl);
+			initRequest.setCannedACL(objectAcl);
+		}
+
+		AccessControlList acl = exchange.getIn().getHeader(S3Constants.ACL, AccessControlList.class);
+		if (acl != null) {
+			// note: if cannedacl and acl are both specified the last one will
+			// be used. refer to
+			// PutObjectRequest#setAccessControlList for more details
+			initRequest.setAccessControlList(acl);
+		}
+
+		LOG.trace("Initiating multipart upload [{}] from exchange [{}]...", initRequest, exchange);
+
+		final InitiateMultipartUploadResult initResponse = getEndpoint().getS3Client()
+				.initiateMultipartUpload(initRequest);
+		final long contentLength = objectMetadata.getContentLength();
+		final List<PartETag> partETags = new ArrayList<PartETag>();
+		long partSize = getConfiguration().getPartSize();
+		CompleteMultipartUploadResult uploadResult = null;
+
+		long filePosition = 0;
+
+		try {
+			for (int part = 1; filePosition < contentLength; part++) {
```

Comment:
I'm not sure what the purpose of the part variable is here.  Is there a reason for using it instead of the filePosition variable as your declared and incremented variable?

A more standard java syntax would be either:
`for (long filePosition = 0; filePosition < contentLength; filePosition++) {`

or...
`long filePosition = 0;`
`int part = 1;`
`while (filePosition < contentLength) {`